### PR TITLE
Rename Compatibility Dates to Compatibility Flags

### PR DESCRIPTION
**Pull request by vicb**
Rename:
- Component: CompatibilityDates -> CompatibilityFlags
- Files
- Collection: compatibility-dates -> compatibility-flags
- Schema: CompatibilityDatesSchema -> CompatibilityFlagsSchema

Update:
- CODEOWNERS
- Redirect: https://developers.cloudflare.com/workers/platform/compatibility-dates.json -> compatibility-flags.json

### Details

Follow up for https://github.com/cloudflare/cloudflare-docs/pull/17900

The name "Compatibility Dates" probably comes from the fact that there was no "Compatibility Flags" section prior ro #17900. However the collection and the component list flags, not dates.

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [x] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

**Comment by vicb** [comment]
Checks:
- https://compatflags.cloudflare-docs-7ou.pages.dev/workers/configuration/compatibility-flags/ content is correct
- https://compatflags.cloudflare-docs-7ou.pages.dev/workers/platform/compatibility-dates.json redirects to https://compatflags.cloudflare-docs-7ou.pages.dev/workers/platform/compatibility-flags.json and the content is correct

/cc @ToriLindsay that should cover what we discussed offline. Thanks for the help.

**Comment by vicb** [comment]
rebased

File: public/_redirects [hunk 1]
```diff
@@ -1449,6 +1449,7 @@
 /workers/platform/triggers/cron-triggers/ /workers/configuration/cron-triggers/ 301
 /workers/platform/cron-triggers/ /workers/configuration/cron-triggers/ 301
 /workers/platform/compatibility-dates/ /workers/configuration/compatibility-dates/ 301
+/workers/platform/compatibility-dates.json /workers/platform/compatibility-flags.json 301
```
Comment (ToriLindsay) [inline public/_redirects:1452]:
```suggestion
/workers/platform/compatibility-dates/ /workers/platform/compatibility-flags/ 301
```
@vicb Just confirming that it shouldn't like this instead? Feel free to close this suggestion if not. Let me know if you're not sure and I'll find out.

**Review by ToriLindsay — APPROVED** [review]
@vicb Thanks for confirming that everything works!

File: public/_redirects (hunk 1 above)
Comment (vicb) [reply public/_redirects:1452]: No.
The only path changed by this PR is the json file, see https://github.com/cloudflare/cloudflare-docs/pull/18176#issuecomment-2477040341

**Review by KianNH — APPROVED** [review]

**Comment by vicb** [comment]
Thanks for your help @ToriLindsay !

**Pull request merged**